src/framework/mpas_stream_manager.F

```diff
@@ -4495,6 +4497,123 @@ subroutine update_stream(stream, allFields, timeLevelIn, mgLevelIn, ierr) !{{{
+        real(kind=RKIND)    :: tmp_real
+        character :: tmp_char
+        integer(kind=I8KIND) :: max_field_bytes, field_bytes
+        integer(kind=I8KIND), parameter :: int_size = c_sizeof(tmp_int)
+        integer(kind=I8KIND), parameter :: real_size = c_sizeof(tmp_real)
+        integer(kind=I8KIND), parameter :: char_size = c_sizeof(tmp_char)
+        integer(kind=I8KIND), parameter :: cdf2_max_bytes = 2**32 - 4
+        character(len=StrKIND):: message
+
+        call mpas_pool_begin_iteration(field_pool)
+        max_field_bytes = 0
+        field_bytes = 0
+        do while ( mpas_pool_get_next_member(field_pool, itr) )
+
+            if (itr % memberType == MPAS_POOL_CONFIG) then
+
+                ! To avoid accidentally matching in case statements below...
+                info % fieldType = -1
+                call mpas_pool_get_field_info(allFields, itr % memberName, info)
+                ! Set time level to read
+                if (info % nTimeLevels >= timeLevelIn) then
+                    timeLevel = timeLevelIn
+                else
+                    timeLevel = 1
+                end if
+                call mpas_log_write('In check_max_var_size, field '//trim(itr % memberName)//' ndims: $i',intArgs=(/info % nDims/))
+                select case (info % fieldType)
+                    case (MPAS_POOL_REAL)
+                        select case (info % nDims)
+                            case (0)
+                                field_size = 1
+                            case (1)
+                                call mpas_pool_get_field(allFields, itr % memberName, real1d, timeLevel)
+                                field_size = global_dim_size(real1d % block, real1d % dimNames, real1d % dimSizes, info % nDims)
+                            case (2)
+                                call mpas_pool_get_field(allFields, itr % memberName, real2d, timeLevel)
+                                field_size = global_dim_size(real2d % block, real2d % dimNames, real2d % dimSizes, info % nDims)
+                            case (3)
+                                call mpas_pool_get_field(allFields, itr % memberName, real3d, timeLevel)
+                                field_size = global_dim_size(real3d % block, real3d % dimNames, real3d % dimSizes, info % nDims)
+                            case (4)
+                                call mpas_pool_get_field(allFields, itr % memberName, real4d, timeLevel)
+                                field_size = global_dim_size(real4d % block, real4d % dimNames, real4d % dimSizes, info % nDims)
+                            case (5)
+                                call mpas_pool_get_field(allFields, itr % memberName, real5d, timeLevel)
+                                field_size = global_dim_size(real5d % block, real5d % dimNames, real5d % dimSizes, info % nDims)
+                        end select
+                        field_bytes = int(field_size, kind=I8KIND) * real_size
```

Comment:
Perhaps `field_size` should be declared as an `I8KIND`, and this type conversion should be eliminated.

Reply:
Isn't there a type conversion involved whether it's in this function or inside `global_dim_size `. Not sure if `mpas_pool_get_dimension` can work with I8KIND int pointers?

Reply:
I've now changed this. The only type conversion happens in `global_dim_size `, due to `mpas_pool_get_dimension ` forcing a regular int. And I've changed to `mpas_dmpar_sum_int8` now